pkg/types/relayer.go

```diff
@@ -0,0 +1,79 @@
+package types
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/google/uuid"
+)
+
+// PluginArgs are the args required to create any OCR2 plugin components.
+// It's possible that the plugin config might actually be different
+// per relay type, so we pass the config directly through.
+type PluginArgs struct {
+	TransmitterID string
+	PluginConfig  []byte
+}
+
+type RelayArgs struct {
+	ExternalJobID uuid.UUID
+	JobID         int32
+	ContractID    string
+	New           bool // Whether this is a first time job add.
+	RelayConfig   []byte
+}
+
+type PluginRelayer interface {
+	NewRelayer(ctx context.Context, config string, keystore Keystore) (Relayer, error)
+}
+
+// Relayer extends [types.Relayer] and includes [context.Context]s.
+type Relayer interface {
+	Service
+
+	NewConfigProvider(context.Context, RelayArgs) (ConfigProvider, error)
+	NewMedianProvider(context.Context, RelayArgs, PluginArgs) (MedianProvider, error)
+	NewMercuryProvider(context.Context, RelayArgs, PluginArgs) (MercuryProvider, error)
+	NewFunctionsProvider(context.Context, RelayArgs, PluginArgs) (FunctionsProvider, error)
+
+	ChainStatus(ctx context.Context, id string) (ChainStatus, error)
+	ChainStatuses(ctx context.Context, offset, limit int) (chains []ChainStatus, count int, err error)
+
+	NodeStatuses(ctx context.Context, offset, limit int, chainIDs ...string) (nodes []NodeStatus, count int, err error)
+
+	SendTx(ctx context.Context, chainID, from, to string, amount *big.Int, balanceCheck bool) error
+}
+
+type ChainStatus struct {
+	ID      string
+	Enabled bool
+	Config  string // TOML
+}
+
+type NodeStatus struct {
```

Comment:
can you remind me why this is defined here as opposed to in protobuf? are we shadowing definitions? 

or am i thinking about it in an inverted way? is protobuf a hidden implementation detail of this repo?

Reply:
We haven't had any first class protos so far - they are all mirroring some Go type. We could start using them more directly to save on the conversions though I suppose.

Reply:
Actually, we'll have to do this in order to realize compatibility with GRPC services that add new parameters over time. Otherwise importers will have to react to type changes even when they do not interact with the new fields.

Reply:
FWIW if we settle for RelayerCtx in the short term (or just don't move that one yet?), then this is not a breaking change.

Reply:
Pushed again to leave only the Relayer and PluginRelayer interfaces behind in LOOP (as well as aliases for the others). This should maintain compatability, and allow for gradually updating all the repos to use `package types` instead of the now deprecated aliases.

Reply:
Sweet, that fixed some CI.